File: examples/javascript/ht9170.js
/*
* Author: Zion Orent <zorent@ics.com>
* Copyright (c) 2015 Intel Corporation.
*
* This program and the accompanying materials are made available under the
* terms of the The MIT License which is available at
* https://opensource.org/licenses/MIT.
*
* SPDX-License-Identifier: MIT
*/
var ht9170_dtmf_lib = require("jsupm_ht9170");

// Instantiate a DTMF decoder
var dtmf_obj = new ht9170_dtmf_lib.HT9170(12, 11, 10, 9, 8);

// Now we just spin in a loop, sleeping every 100ms, checking to see
// if a digit is available.  If so, we decode and print the digit,
// and continue looping.
var myInterval = setInterval(function()
{
	if (dtmf_obj.digitReady())
	{
		console.log("Got DTMF code: " + dtmf_obj.decodeDigit());
		// now spin until digitReady() goes false again
		while (dtmf.digitReady())
			;
	}
}, 100);

// When exiting: clear interval, and print message
process.on('SIGINT', function()
{
  clearInterval(myInterval);
	dtmf_obj = null;
	ht9170_dtmf_lib.cleanUp();
	ht9170_dtmf_lib = null;
	console.log("Exiting");
	process.exit(0);
});

File: examples/c++/ht9170.cxx
/*
 * Author: Jon Trulson <jtrulson@ics.com>
 * Copyright (c) 2015 Intel Corporation.
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */

#include <iostream>
#include <signal.h>

#include "ht9170.hpp"
#include "upm_utilities.h"

using namespace std;

int shouldRun = true;

void
sig_handler(int signo)
{
    if (signo == SIGINT)
        shouldRun = false;
}

int
main()
{
    signal(SIGINT, sig_handler);
    //! [Interesting]

    // Instantiate a DTMF decoder
    upm::HT9170 dtmf(12, 11, 10, 9, 8);

    // Now we just spin in a loop, upm_delaying every 100ms, checking to see
    // if a digit is available.  If so, we decode and print the digit,
    // and continue looping.

    while (shouldRun) {
        if (dtmf.digitReady()) {
            cout << "Got DTMF code: " << dtmf.decodeDigit() << endl;
            // now spin until digitReady() goes false again
            while (dtmf.digitReady())
                ;
        }

        upm_delay_us(100000);
    }

    //! [Interesting]

    cout << "Exiting..." << endl;

    return 0;
}
